daemon/container_linux.go

```diff
@@ -0,0 +1,34 @@
+//+build !windows
+
+package daemon
+
+import (
+	containertypes "github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/container"
+)
+
+func (daemon *Daemon) setSecurityOptions(container *container.Container, hostConfig *containertypes.HostConfig) error {
+	container.Lock()
+	defer container.Unlock()
+	if err := parseSecurityOpt(container, hostConfig); err != nil {
+		return err
+	}
+	if daemon.apparmorEnabled && !hostConfig.Privileged {
+		if container.AppArmorProfile == "" {
+			container.AppArmorProfile = "docker-default"
+		}
+	} else {
+		container.AppArmorProfile = "unconfined"
+	}
+	return nil
+}
+
+func (daemon *Daemon) fixSecurityOptions(container *container.Container) {
+	if daemon.apparmorEnabled && !container.HostConfig.Privileged {
+		if container.AppArmorProfile == "unconfined" {
+			container.AppArmorProfile = defaultApparmorProfile
```

Comment:
That is actually an error. I was trying to restore "docker-default" to the unconfined containers if the containers were created after apparmor was enabled but i didn't notice the effect of enabling apparmor to the containers with --security-opt app-armor:unconfined.

I'm gonna think on how manage this correctly.